src/fdc/DiskManipulator.cc

```diff
@@ -370,7 +380,13 @@ void DiskManipulator::tabCompletion(std::vector<string>& tokens) const
 			// if it has partitions then we also add the partition
 			// numbers to the autocompletion
 			if (auto* disk = d.drive->getSectorAccessibleDisk()) {
-				for (unsigned i = 1; i <= MAX_PARTITIONS; ++i) {
+				for (unsigned i = 1; ; ++i) {
```

Comment:
Detail: in openMSX we usually fill-in all three fields in a for loop. In this case that's `true` instead of an empty condition  (and we use `/**/` if the other two fields would be empty).